exec/cnex/dictionary.c

```diff
@@ -68,6 +68,7 @@ void dictionary_addDictionaryEntry(Dictionary *self, struct tagTString *key, str
     self->data[self->len].key = key;
     self->data[self->len].value = value;
     self->len++;
+    return self->len;
```

Comment:
Are you sure this is the right value to return here? Not `self->len - 1`?

Reply:
I....  I do believe you are right.  After looking at this, yes, I'm incrementing the length, and THEN returning the index, but the index has just been incremented.  So, I"ve now changed this to return the self->len - 1.  This should do the trick.